server/gfmd/db_ldap.c

```diff
@@ -3264,6 +3264,162 @@ gfarm_ldap_quota_dir_load(void *closure,
 	return (GFARM_ERR_FUNCTION_NOT_IMPLEMENTED);
 }
 
+/**********************************************************************/
+
+static char *gfarm_ldap_user_auth_make_dn(void *vkey);
+static gfarm_error_t gfarm_ldap_user_auth_set_field(void *info,
+	char *attribute, char **vals);
+
+struct gfarm_ldap_user_auth_key {
+	const char *username;
+	const char *auth_method;
+};
+
+static const struct gfarm_ldap_generic_info_ops gfarm_ldap_user_auth_ops = {
+	&gfarm_base_user_auth_ops,
+	"(objectclass=GFarmUserAuth)",
+	"username=%s, auth_method=%s, %s",
+	gfarm_ldap_user_info_make_dn,
```

Comment:
Please replace gfarm_ldap_user_info_make_dn by gfarm_ldap_user_auth_make_dn,
compilation shows the following warning:
```
./db_ldap.c:1567:1: warning: 'gfarm_ldap_user_auth_make_dn' defined but not used [-Wunused-function]
 gfarm_ldap_user_auth_make_dn(void *vkey)
```